sw/device/boot_rom/rom_crt.S

```diff
@@ -75,18 +81,39 @@ _reset_start:
 _start:
   .globl _start
 
+  // Enable entropy complex - this is not the full enable
+  li   a0, TOP_EARLGREY_ENTROPY_SRC_BASE_ADDR
+  li   t0, 0x2
+  sw   t0, ENTROPY_SRC_CONF_REG_OFFSET(a0)
+  li   a0, TOP_EARLGREY_CSRNG_BASE_ADDR
+  li   t0, 0x1
+  sw   t0, CSRNG_CTRL_REG_OFFSET(a0)
+  li   a0, TOP_EARLGREY_EDN0_BASE_ADDR
+  li   t0, 0x1
+  sw   t0, EDN_CTRL_REG_OFFSET(a0)
+
+  // Request memory scrambling and init
+  li   a0, TOP_EARLGREY_SRAM_CTRL_MAIN_BASE_ADDR
+  li   t0, 1
+  slli t1, t0, SRAM_CTRL_CTRL_RENEW_SCR_KEY_BIT
+  mv   a1, t1
+  slli t1, t0, SRAM_CTRL_CTRL_INIT_BIT
+  or   a1, a1, t1
+  sw   a1, SRAM_CTRL_CTRL_REG_OFFSET(a0)
+
   // Zero out the `.bss` segment.
   la   a0, _bss_start
   la   a1, _bss_end
   call crt_section_clear
 
+  // This zero out stack function is now handled by the hardware memory init from above
```

Comment:
According to https://docs.opentitan.org/hw/ip/sram_ctrl/doc/ Programmers Guide/Initialization:

The memory contents are not needed anymore. Requesting a key implicitly wipes all data in the SRAM.

Does this mean we can also eliminate explicitly zeroing `.bss` as well (all of L104 - L116)?  If so, I'd just delete the code and leave a comment to the effect of

""
Initializing memory scrambling also zeros out all of RAM.  Explicit zeroing of the .bss and stack sections is not required.
"""

Reply:
+1

Reply:
We still need to zero `.bss` after scrambling is enabled because we want to read back actual zeros ([@tjaychen's comment](https://github.com/lowRISC/opentitan/pull/5714#issuecomment-803195360)). I think the part for stack (currently commented out) can just be removed.

Reply:
o just to be clear i left the bss initialization.  I only removed the stack one. (unless i did something stupid and removed both?)

Reply:
I think Alphan is recommending you remove the stack-clearing code entirely, rather than leaving it commented out. If so, I agree. I'm fairly confident the C ABI assumes stack memory below the current stack frame is uninitialized.

Reply:
o sorry, i just saw this message. Let me remove it with a separate PR.
Yeah I kept it there just in case anyone was going to ask me to revert hehe.